AnnoyancesFilter/Other/sections/self-promo.txt

```diff
@@ -176,6 +176,34 @@ habr.com##ul.tm-article-list-block__list > li.tm-article-list-block__item:has(>
 habr.com##.tm-article-card-list > li.tm-article-card-list__item:has(> div.tm-promo-post-link)
 habr.com#?#section.tm-block_spacing-around:has(> header > div > h2:contains(Ближайшие события))
 !
+! 
+! linkedin.com
+www.linkedin.com##span:has-text(Promoted):upward(8)
+www.linkedin.com##span:has-text(Promoted)
+www.linkedin.com##li:has-text(Promoted)
+www.linkedin.com##span:has-text(Suggested):upward(8)
+www.linkedin.com##span:has-text(Suggested)
+www.linkedin.com##li:has-text(Suggested)
+www.linkedin.com##span:has-text(Recommended):upward(8)
+www.linkedin.com##span:has-text(Recommended) 
+www.linkedin.com##li:has-text(Recommended)
+www.linkedin.com##span:has-text(orgeschlagen):upward(8)
+www.linkedin.com##span:has-text(orgeschlagen)
+www.linkedin.com##li:has-text(orgeschlagen)
+www.linkedin.com##span:has-text(obs recommende):upward(8)
+www.linkedin.com##span:has-text(obs recommende)
+www.linkedin.com##span:has-text(esponsert):upward(8)
+www.linkedin.com##span:has-text(esponsert)
+www.linkedin.com##li:has-text(esponsert)
+www.linkedin.com##li:has-text(obs recommende)
+www.linkedin.com##.premium-upsell-link
+www.linkedin.com##div.artdeco-card.overflow-hidden:has(a[class*="premium-upsell"])
+www.linkedin.com##section.ad-banner-container
+www.linkedin.com##div.mn-sales-navigator-upsell.artdeco-card
+www.linkedin.com##section.artdeco-card:has(div[class^="jobs-home-upsell-"])
+www.linkedin.com##div:has(a[class^="premium-upsell-link"])  
+www.linkedin.com##span.global-nav__secondary-premium-cta-text:upward(1)
```

Comment:
Better use regexp, and `:has` instead of `:upward`. These rules has low performance.
Anlso need examples of ads - I can't reproduce.